### PR TITLE
Upgrade to ocamlformat.0.21.0

### PR DESCRIPTION
**Pull request by gpetiot**
This is a preview of the formatting of your project codebase with the upcoming `ocamlformat.0.21.0`. Note that this package is not yet released in opam (so the linting/formatting task of your CI won't pass), the formatting might also change slightly in the released version as we try to integrate feedback from our users.

The changes on this project are due to:
- the list of arguments of function applications break on "complex" argument, an argument being composed of a variant is not considered being "complex" anymore.

As always, we are happy to hear your feedback!  Thank you for using ocamlformat!

### CODE REVIEW

File: test/irmin-graphql/common.mli [hunk 1]
```diff
@@ -59,12 +59,16 @@ type query
     For example, the following query returns the latest commit hash for the
     [main] branch:
 
-    {[ query (func "main" (field "hash")) ]}
+    {[
```
Comment (tomjridge) [inline test/irmin-graphql/common.mli:62]: What's the logic behind ocamlformat inserting blank lines at the beginning and end of codeblocks?

Comment (tomjridge) [reply test/irmin-graphql/common.mli:62]: Thanks! So this is some extra refinement of ocamldoc syntax implemented by odoc. Is the restriction documented anywhere? (Presumably the blank lines are not considered part of the actual code?)

Comment (gpetiot) [reply test/irmin-graphql/common.mli:62]: I don't think it is documented anywhere, it is more of a practical choice for the formatting, so we don't have to check whether the code is big enough to break or if it was initially breaking and whether we should preserve it, we break everytime.

It looks like `odoc-parser` is trimming the leading and trailing blanklines in the code block.

**Comment by icristescu** [comment]
Thanks!

**Pull request merged**